global/scripts/processingmodecheck.shlib

```diff
@@ -0,0 +1,54 @@
+#!/bin/echo This script should not run directly:
+
+# Functions for checking compliance with selected MPP mode for HCP Pipelines
+
+# --- Import log.shlib 
+
+if [[ "$(type -t log_Err_Abort)" != "function" ]]
+then
+    source "$HCPPIPEDIR/global/scripts/log.shlib" "$@"
+fi
+
+
+check_mode_compliance()
+{
+
+    # --- Define variables
+
+    local ProcessingMode="$1"
+    local Compliance="$2"
+    local ComplianceMsg="$3"
+
+    # --- Initial information
+
+    if [ "${ProcessingMode}" = "LegacyStyleData" ]; then
+      log_Msg "LegacyStyleData mode requested."
+      log_Msg "NOTICE: You are using a mode that enables processing of acquisitions that do not"
+      log_Msg "        conform to the HCP specification as described in Glasser et al. (2013)!"
+      log_Msg "        Be aware that if the HCP requirements are not met, the level of data quality"
+      log_Msg "        can not be guaranteed and the Glasser et al. (2013) paper should not be used"
+      log_Msg "        in support of this workflow. A manuscript with comprehensive evaluation for"
+      log_Msg "        the Extended MPP workflow is in active preparation and should be appropriately"
+      log_Msg "        cited when published."
+      log_Msg "        "
+    fi
+
+    # -- Final evaluation
+
+    if [ "${ProcessingMode}" = "LegacyStyleData" ]; then
+      if [ "${Compliance}" = "HCPStyleData" ]; then
+        log_Warn "All conditions for the use of --processing-mode=HCPStyleData are met. Consider using that mode (or equivalently, simply leave --processing-mode empty) instead of --processing-mode=LegacyStyleData."
```

Comment:
Candidate for splitting into two lines.

Reply:
I'm not so sure about this one, it doesn't seem great for it to appear like:

```
Wed Oct  9 16:14:40 CDT 2019:PreFreeSurfer.sh: WARNING: All conditions for the use of ...
Wed Oct  9 16:14:40 CDT 2019:PreFreeSurfer.sh: WARNING: Consider using that mode...
```

Repeating the date and "WARNING" is a bit less clear.  If you wanted to repeat the date, then the second could be a log_Msg, but I'd rather not repeat the date or introduce any artificial separation between these clearly related sentences, so I vote for leaving this one as-is.

Reply:
If you really want a manual line break, it is possible, though a bit ugly, to do something like this:

```
...
            log_Warn "All conditions for the use of --processing-mode=HCPStyleData are met.
Consider using that mode (or equivalently, simply leave --processing-mode empty) instead of --processing-mode=LegacyStyleData."
```

I still vote leave it as-is.

Reply:
Ah, I see the above blocks of log_Msg are doing some manual width control and will repeat the date and tool name on each line - I'm not sure if that kind of manual line splitting is how we really want to do long log messages, but I can live with it for now.

Reply:
Ok. Did it the same here.